felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -2527,37 +2583,58 @@ func (d *bpfEndpointManagerDataplane) applyPolicyToWeps(
 	return ingressAP, egressAP, errors.Join(ingressErr, egressErr)
 }
 
-func (d *bpfEndpointManagerDataplane) applyPolicyToDataIface(
+func (d *bpfEndpointManagerDataplane) attachTCProgramToDataIface(
 	ifaceName string,
 	ep *proto.HostEndpoint,
 	state *bpfInterfaceState,
-	ap *tc.AttachPoint,
-	apxdp *xdp.AttachPoint,
-) (*tc.AttachPoint, *tc.AttachPoint, *xdp.AttachPoint, error) {
+	ap *tc.AttachPoint) (*tc.AttachPoint, *tc.AttachPoint, error) {
 	ingressAttachPoint := *ap
 	egressAttachPoint := *ap
-	xdpAttachPoint := *apxdp
 
+	var ingressErr, egressErr error
 	var parallelWG sync.WaitGroup
 	var ingressAP, egressAP *tc.AttachPoint
-	var xdpAP *xdp.AttachPoint
-	var ingressErr, egressErr, xdpErr error
-
-	parallelWG.Add(2)
```

Comment:
cann't you just have `parallelWG.Add(1)` infron of each goroutine here and the other (xdp) covered by `if ifaceType != IfaceTypeBond {` without creating the other new function which only groups attaching TC? They that function waits for both tc and then for xdp, while here it waited for tc and xdp at the same time :thinking: